tests/backends/managers/test_profile_access.py

```diff
@@ -0,0 +1,363 @@
+    methods, we just need to check that all the relevant information is passed and shown on the
+    error message.
+    """
+    pass_message = 'This message should be on the error value'
+    file_content = 'This is the file content'
+    file_stem = '123456'
+
+    tempfile = Path(file_stem + '.txt')
+    tempfile.write_text(file_content, encoding='utf-8')
+
+    def mock_get_tracking_files(*args, **kwargs):
+        """Mock of _raise_if_locked."""
+        return [tempfile]
+
+    monkeypatch.setattr(profile_access_manager, '_get_tracking_files', mock_get_tracking_files)
+
+    with pytest.raises(LockingProfileError) as exc:
+        profile_access_manager._raise_if_active(pass_message)
+    error_message = str(exc.value)
+    assert pass_message in error_message
+    assert tempfile.stem in error_message
+    assert file_content in error_message
+
+    with pytest.raises(LockedProfileError) as exc:
+        profile_access_manager._raise_if_locked(pass_message)
+    error_message = str(exc.value)
+    assert pass_message in error_message
+    assert tempfile.stem in error_message
+
+    tempfile.unlink()
+
+
+def test_clear_locks(profile_access_manager):
+    """Tests the `test_clear_locks` method.
+
+    For this it creates an artificial lock file and uses the `is_locked` method
+    (already tested elsewhere) to check if the profile is recognized as being
+    locked in the different steps.
+    """
+    records_dir = profile_access_manager._dirpath_records
+    lockfile = records_dir / '1234567890.lock'
+
+    assert not profile_access_manager.is_locked()
+    lockfile.touch()
+    assert profile_access_manager.is_locked()
+    profile_access_manager.clear_locks()
+    assert not profile_access_manager.is_locked()
+
+
+def test_clear_stale_pid_files(profile_access_manager):
+    """Tests the `_clear_stale_pid_files` method."""
+    records_dir = profile_access_manager._dirpath_records
+    fake_filename = '1234567890.pid'
+    fake_file = records_dir / fake_filename
+    fake_file.touch()
+
+    assert fake_filename in listdir(records_dir)
+    profile_access_manager._clear_stale_pid_files()
+    assert fake_filename not in listdir(records_dir)
+
+
+###########################################################################
+# SEMI INTEGRATION TESTS
+#
+# This second section contains more complex tests that go through more
+# than one method to test the cohesion of the whole class when working
+# with actual processes. It is therefore also dependant on the call to
+# the class in:
+#
+#    >   aiida.manage.manager::Manager._load_backend
+#
+# Moreover, they also require the use of a separate construct to keep
+# track of processes accessing aiida profiles with ease (TestProcess).
+#
+# This is separated from the more simple unit tests for each method.
+#
+###########################################################################
+
+
+class TestProcess():
```

Comment:
Suggest renaming this, since with the current name `pytest` think it is an actual class of tests and will emit a warning
```suggestion
class MockProcess():
```

Reply:
Ah, I didn't notice this warning. Thanks! Is changed now.